In [1]:
#Import Libraries
import pandas as pd
import numpy as numpy
import matplotlib.pyplot as pyplot
import seaborn as sns
import warnings


warnings.filterwarnings('ignore')
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', None)

In [2]:
# Load LinkedIn Data
df = pd.read_csv('./data/Post Performance (Stanbic IBTC) January 1, 2013 - July 13, 2023_LinkedIn.csv', low_memory=False)

In [3]:
# Check data dimensions
print('Rows: ', df.shape[0])
print('Columns: ', df.shape[1])

Rows:  7760
Columns:  147


In [4]:
# Examine first two observations
df.head(2)

Date              Post ID   Network Post Type Content Type  \
0  7/13/2023 12:03 pm  7085212016608456704  LinkedIn      Post        Photo   
1   7/12/2023 8:00 pm  7084969706025009152  LinkedIn      Post         Text   

        Profile        Sent by  \
0  Stanbic IBTC  Blessing Ubah   
1  Stanbic IBTC  Blessing Ubah   

                                                                Link  \
0  https://linkedin.com/feed/update/urn:li:share:7085212016608456704   
1  https://linkedin.com/feed/update/urn:li:share:7084969706025009152   

                                                                                                                                                                                                                                                                                       Post  \
0  Your money should not be a gamble. Before you invest, do your own personal research on the historical performance of the stocks and ensure you invest with a credible firm like Stanbic IBTC Stockbrokers.\nSend an email to CallMe@stanbicibtc.com to get started.\n#ITCANBE\n#Invest\n   
1                                                                                                                                                                                             What skill (s) do you have, that no one ever taught you? \nShare in the comments!\n#ITCANBE\n   

  Linked Content Impressions Organic Impressions  Viral Impressions  \
0            NaN         NaN                 NaN                NaN   
1            NaN         NaN                 NaN                NaN   

   Non-viral Impressions  Paid Impressions  Fan Impressions  \
0                    NaN               NaN              NaN   
1                    NaN               NaN              NaN   

   Fan Organic Impressions  Fan Paid Impressions  Non-fan Impressions  \
0                      NaN                   NaN                  NaN   
1                      NaN                   NaN                  NaN   

   Non-fan Organic Impressions  Non-fan Paid Impressions  Reach  \
0                          NaN                       NaN    NaN   
1                          NaN                       NaN    NaN   

   Organic Reach  Viral Reach  Non-viral Reach  Paid Reach  Fan Reach  \
0            NaN          NaN              NaN         NaN        NaN   
1            NaN          NaN              NaN         NaN        NaN   

   Fan Paid Reach  Potential Reach Engagement Rate (per Impression)  \
0             NaN              NaN                              NaN   
1             NaN              NaN                              NaN   

   Engagement Rate (per Reach) Engagements  Reactions  Likes  Dislikes  \
0                          NaN         NaN        NaN    NaN       NaN   
1                          NaN         NaN        NaN    NaN       NaN   

   Love Reactions  Haha Reactions  Wow Reactions  Sad Reactions  \
0             NaN             NaN            NaN            NaN   
1             NaN             NaN            NaN            NaN   

   Angry Reactions  Comments  Shares  Saves Click-Through Rate  \
0              NaN       NaN     NaN    NaN                NaN   
1              NaN       NaN     NaN    NaN                NaN   

  Post Link Clicks  Other Post Clicks Post Clicks (All)  Post Media Clicks  \
0              NaN                NaN               NaN                NaN   
1              NaN                NaN               NaN                NaN   

   Post Hashtag Clicks  Post Detail Expand Clicks  Profile Clicks  \
0                  NaN                        NaN             NaN   
1                  NaN                        NaN             NaN   

   Post Photo View Clicks  Post Video Play Clicks  Other Engagements  Answers  \
0                     NaN                     NaN                NaN      NaN   
1                     NaN                     NaN                NaN      NaN   

   App Engagements  App In

## Data Cleaning

### Drop Columns

The plan is to drop columns in which atleast 80% of the data are null values.

These columns cannot be used in the analysis

In [5]:
# Return columns where null values are >= 80%
null_columns = list(df.columns[df.isnull().sum() >= 0.8*len(df)])

In [6]:
# Get the columns that are not completely null
not_null_columns = df.columns[~df.columns.isin(null_columns)]

# Create a new dataframe with only the non-null columns
df_not_null = df[not_null_columns]

In [7]:
# Check new dataframe dimensions
print('Rows: ', df_not_null.shape[0])
print('Columns: ', df_not_null.shape[1])

Rows:  7760
Columns:  21


In [8]:
# Examine first two observations
df_not_null.head(2)

Date              Post ID   Network Post Type Content Type  \
0  7/13/2023 12:03 pm  7085212016608456704  LinkedIn      Post        Photo   
1   7/12/2023 8:00 pm  7084969706025009152  LinkedIn      Post         Text   

        Profile        Sent by  \
0  Stanbic IBTC  Blessing Ubah   
1  Stanbic IBTC  Blessing Ubah   

                                                                Link  \
0  https://linkedin.com/feed/update/urn:li:share:7085212016608456704   
1  https://linkedin.com/feed/update/urn:li:share:7084969706025009152   

                                                                                                                                                                                                                                                                                       Post  \
0  Your money should not be a gamble. Before you invest, do your own personal research on the historical performance of the stocks and ensure you invest with a credible firm like Stanbic IBTC Stockbrokers.\nSend an email to CallMe@stanbicibtc.com to get started.\n#ITCANBE\n#Invest\n   
1                                                                                                                                                                                             What skill (s) do you have, that no one ever taught you? \nShare in the comments!\n#ITCANBE\n   

  Impressions Organic Impressions Engagement Rate (per Impression)  \
0         NaN                 NaN                              NaN   
1         NaN                 NaN                              NaN   

  Engagements  Reactions  Likes  Comments  Shares Click-Through Rate  \
0         NaN        NaN    NaN       NaN     NaN                NaN   
1         NaN        NaN    NaN       NaN     NaN                NaN   

  Post Link Clicks Post Clicks (All)       Tags  
0              NaN               NaN  CIB Posts  
1              NaN               NaN        NaN

In [9]:
# print column names 
print(df_not_null.columns)

Index(['Date', 'Post ID', 'Network', 'Post Type', 'Content Type', 'Profile',
       'Sent by', 'Link', 'Post', 'Impressions', 'Organic Impressions',
       'Engagement Rate (per Impression)', 'Engagements', 'Reactions', 'Likes',
       'Comments', 'Shares', 'Click-Through Rate', 'Post Link Clicks',
       'Post Clicks (All)', 'Tags'],
      dtype='object')


### Quality Issues 

The plan is to examine the data for
* duplicates
* wrong data types
* inconsistent / wrong formatting and
* missing data

In [10]:
# Check duplicates
print('Number of duplicates: ', df_not_null.duplicated().sum())

Number of duplicates:  0


In [11]:
# Examine datatypes
pd.DataFrame(df_not_null.dtypes)

0
Date                               object
Post ID                             int64
Network                            object
Post Type                          object
Content Type                       object
Profile                            object
Sent by                            object
Link                               object
Post                               object
Impressions                        object
Organic Impressions                object
Engagement Rate (per Impression)   object
Engagements                        object
Reactions                         float64
Likes                             float64
Comments                          float64
Shares                            float64
Click-Through Rate                 object
Post Link Clicks                   object
Post Clicks (All)                  object
Tags                               object

In [12]:
# Check missing values
pd.DataFrame(df_not_null.isna().sum()).sort_values(by=0, ascending=False)

0
Tags                              6120
Engagement Rate (per Impression)  1428
Post Clicks (All)                 1428
Post Link Clicks                  1428
Click-Through Rate                1428
Shares                            1428
Comments                          1428
Likes                             1428
Reactions                         1428
Engagements                       1428
Organic Impressions               1428
Impressions                       1428
Post                                31
Link                                 4
Post ID                              0
Sent by                              0
Profile                              0
Content Type                         0
Post Type                            0
Network                              0
Date                                 0

#### List of Quality Issues

* **14** columns have missing values
* 'Engagement Rate (per Impression)' & 'Click-Through Rate' columns have % sign values
* Several columns have the wrong datatype
    * 8 columns are wrongly labeled as strings instead of numerics.
    * The Date column should be a datetime object.
    * 'Sent by' and 'Content Type' columns should be categorical.
* Several columns have values are formatted with ',' e.g. '200,000'. This format cannot be converted to numeric without issues. They must be treated while converting datatypes.
* 'Sent by' column has blank values (i.e missing values are filled with space).

---

#### Fixing Missing Data Issues

In [13]:
# Check rows where Impressions are null.
df_not_null[df_not_null['Impressions'].isna()].head()

Date              Post ID   Network Post Type Content Type  \
0   7/13/2023 12:03 pm  7085212016608456704  LinkedIn      Post        Photo   
1    7/12/2023 8:00 pm  7084969706025009152  LinkedIn      Post         Text   
33   7/5/2023 11:02 am  7082297618676031489  LinkedIn      Post        Video   
42    7/3/2023 1:23 pm  7081608447019876352  LinkedIn      Post        Photo   
45   7/3/2023 11:58 am  7081587121429389313  LinkedIn      Post        Photo   

         Profile         Sent by  \
0   Stanbic IBTC   Blessing Ubah   
1   Stanbic IBTC   Blessing Ubah   
33  Stanbic IBTC                   
42  Stanbic IBTC   Blessing Ubah   
45  Stanbic IBTC  Aramide Salami   

                                                                   Link  \
0     https://linkedin.com/feed/update/urn:li:share:7085212016608456704   
1     https://linkedin.com/feed/update/urn:li:share:7084969706025009152   
33  https://linkedin.com/feed/update/urn:li:ugcPost:7082297618676031489   
42  https://linkedin.com/feed/update/urn:li:ugcPost:7081608447019876352   
45  https://linkedin.com/feed/update/urn:li:ugcPost:7081587121429389313   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Post  \
0                                                                                                                                                                                                                                                                                                                                                                             Your money should not be a gamble. Before you invest, do your own personal research on the historical performance of the stocks and ensure you invest with a credible firm like Stanbic IBTC Stockbrokers.\nSend an email to CallMe@stanbicibtc.com to get started.\n#ITCANBE\n#Invest\n   
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        What skill (s) do you have, that no one ever taught you? \nShare in the comments!\n#ITCANBE\n   
33                                                                                                                                                                                                                                                                                                                                                                                                                      Ladies At The Table Series (LATTES) 3.0\n\nHey ladies, are you prepared for an experience that will fuel your determination and ignite the fire within you?\r\nWatch the 2023 edition of the Ladies At The Table Empowerment Series (LATTES).    
42  Driving growth beyond borders is what we do.\nA selection of our clients were at the third biennial China-Africa Economic and Trade Expo (CAETE) through our parent group, Standard Bank, to exhibit their produce such as Soy Beans, Sorghum, Millet, Cashew nuts and Se

##### Note

11 columns have missing data corresponding to deleted posts (1428 observations).

These rows will be deleted.

In [14]:
# Remove rows where Impressions are null.
df_not_null = df_not_null[~df_not_null['Impressions'].isna()]

In [15]:
# Check missing values
pd.DataFrame(df_not_null.isna().sum()).sort_values(by=0, ascending=False).head()

0
Tags                              4738
Post                                31
Engagement Rate (per Impression)     0
Post Clicks (All)                    0
Post Link Clicks                     0

---

In [16]:
# Examine missing posts
df_not_null[df_not_null['Post'].isna()][['Post', 'Link', 'Content Type']]

Post  \
151   NaN   
164   NaN   
247   NaN   
324   NaN   
373   NaN   
448   NaN   
449   NaN   
540   NaN   
592   NaN   
600   NaN   
981   NaN   
1494  NaN   
1917  NaN   
1918  NaN   
2142  NaN   
2361  NaN   
2419  NaN   
2459  NaN   
2689  NaN   
2722  NaN   
2734  NaN   
2762  NaN   
2967  NaN   
2971  NaN   
2976  NaN   
2977  NaN   
3258  NaN   
3463  NaN   
4376  NaN   
4384  NaN   
6108  NaN   

                                                                      Link  \
151    https://linkedin.com/feed/update/urn:li:ugcPost:7071536783951376385   
164    https://linkedin.com/feed/update/urn:li:ugcPost:7070036706661478400   
247      https://linkedin.com/feed/update/urn:li:share:7061594808468873216   
324    https://linkedin.com/feed/update/urn:li:ugcPost:7056211882893819904   
373    https://linkedin.com/feed/update/urn:li:ugcPost:7052202991726075906   
448    https://linkedin.com/feed/update/urn:li:ugcPost:7046169903757701120   
449    https://linkedin.com/feed/update/urn:li:ugcPost:7046083981007568896   
540    https://linkedin.com/feed/update/urn:li:ugcPost:7034810890952552448   
592    https://linkedin.com/feed/update/urn:li:ugcPost:7028808107048808448   
600    https://linkedin.com/feed/update/urn:li:ugcPost:7028442101914714112   
981   https://linkedin.com/feed/update/urn:li:activity:6991422141845958656   
1494  https://linkedin.com/feed/update/urn:li:activity:6943942410817142784   
1917   https://linkedin.com/feed/update/urn:li:ugcPost:6909846171075129344   
1918   https://linkedin.com/feed/update/urn:li:ugcPost:6909835347170340865   
2142  https://linkedin.com/feed/update/urn:li:activity:6894612284560531456   
2361   https://linkedin.com/feed/update/urn:li:ugcPost:6872555469798473729   
2419  https://linkedin.com/feed/update/urn:li:activity:6867780427776610304   
2459  https://linkedin.com/feed/update/urn:li:activity:6865262851133603840   
2689   https://linkedin.com/feed/update/urn:li:ugcPost:6846480725894590465   
2722  https://linkedin.com/feed/update/urn:li:activity:6844224048570871808   
2734  https://linkedin.com/feed/update/urn:li:activity:6843496410264961024   
2762  https://linkedin.com/feed/update/urn:li:activity:6841675759807942658   
2967  https://linkedin.com/feed/update/urn:li:activity:6819236219675799552   
2971  https://linkedin.com/feed/update/urn:li:activity:6818897882947563520   
2976  https://linkedin.com/feed/update/urn:li:activity:6818551453112643584   
2977  https://linkedin.com/feed/update/urn:li:activity:6818551353804124161   
3258  https://linkedin.com/feed/update/urn:li:activity:6789876551195365378   
3463  https://linkedin.com/feed/update/urn:li:activity:6767850615759618049   
4376  https://linkedin.com/feed/update/urn:li:activity:6676041979979157504   
4384  https://linkedin.com/feed/update/urn:li:activity:6674627858641555457   
6108  https://linkedin.com/feed/update/urn:li:activity:6445657238597042177   

     Content Type  
151          Text  
164          Poll  
247          Text  
324          Poll  
373          Text  
448          Text  
449          Text  
540          Text  
592          Text  
600          Text  
981          Text  
1494         Text  
1917        Video  
1918        Video  
2142         Text  
2361        Video  
2419         Text  
2459         Text  
2689        Video  
2722         Text  
2734         Text  
2762         Text  
2967         Text  
2971         Text  
2976         Text  
2977         Text  
3258         Text  
3463         Text  
4376         Text  
4384         Text  
6108        Photo

In [17]:
# Examine content type of missing posts
df_not_null[df_not_null['Post'].isna()][['Content Type']].value_counts()

Content Type
Text            24
Video            4
Poll             2
Photo            1
dtype: int64

##### Note

Posts with missing values do not have text content. 

While most are shown as text above, they are actually embedded links

These values will be imputed with blank space instead.

In [18]:
# fill missing values in 'Post' column with values in ''
df_not_null['Post'] = df_not_null['Post'].fillna('')

In [19]:
# Confirm imputation
df_not_null['Post'].isna().sum()

0

---

In [20]:
#Check posts without tags
df_not_null[df_not_null['Tags'].isna()][['Post','Tags']]

Post  \
4                                                                                                                              Together, we can make a difference and build a brighter tomorrow.  🤝 \n#GlobalUnityForTomorrow\n#WorldPopulationDay\n#ITCANBE\n   
9                                                                                                                                                                   Remember, self-love fuels your external glow. 💙 \n#ITCANBE\n#Selflove\n#MondayMotivation\n   
10                                                             Choose happiness daily to create a more joyful and fulfilling life. Incorporate these tips and make happiness-oriented decisions to cultivate a blissful existence. \n#HappySunday!\n#ITCANBE\n   
12                                                                                                                            Update! Need to swap your FX to naira? We've got you covered! Download the Stanbic IBTC Mobile App now to get started.\n#ITCANBE   
13                                     Don’t let fear of the managing your finances home and abroad stop you from travelling abroad. Do you have question(s) about your money and travelling abroad? Share with us here https://lnkd.in/dcFK-vwH\n\n#ITCANBE\n   
...                                                                                                                                                                                                                                                        ...   
6666  #WealthWednesday We know you love the freedom that comes with self-service. We're pleased to let you know that you have that option when it comes to your pension account too. From your registered number, just text 30388 with the following commands:   
6669                                                                                                                                                #JAMBTip Planning your study time and sticking to it might be one of the best things to do as you prepare.   
6670                                                                                                                                           #TermOfTheDay\nA Permalink is a web address that permanently links to a specific web page, forum or blog entry.   
6671                                       Members of  Stanbic IBTC Asset Management's Executive Committee and Dr. Mansur Ibrahim, Founder of Ahlulbaiti and Sahaba Foundation Nigeria, at the recently held Imaan Fund cheque presentation to the foundation.   
6672                                                                                       #MotivationMonday\r\nWherever you find yourself, complaining and grumbling won’t do you any good but risking to go beyond the norm can. Be motivated by this quote:   

     Tags  
4     NaN  
9     NaN  
10    NaN  
12    NaN  
13    NaN  
...   ...  
6666  NaN  
6669  NaN  
6670  NaN  
6671  NaN  
6672  NaN  

[4738 rows x 2 columns]

In [21]:
#Check posts with tags
df_not_null[df_not_null['Tags'].notna()][['Post','Link', 'Tags']].head()

Post  \
2                 Sade was able to acquire a home loan and make her dream a reality, you too can. Visit https://lnkd.in/dASbfHCz today and let’s help you get that house of your dreams.\n#WealthWednesday\n#ITCANBE\n   
3                                  Safeguard every aspect of your operations with our comprehensive insurance cover. Email insurancesolution@stanbicibtc.com today to secure your organisation's future.\n#ITCANBE\n\n   
5                                                                                                        Hey CEOs, it’s time to say goodbye to complexities and hello to seamless operations.\nStay Tuned! 😉\n#ITCANBE   
6                                       With the Stanbic IBTC SME Online-Bento Redirect Solution, you can easily sort out staff salaries and taxes. Email SpecialisedSectors@stanbicibtc.com to get started.\n#ITCANBE   
7  Embark on your retirement journey with us as your trusted companion guaranteeing a smile at every step along the way. Visit www.stanbicibtcpension.com and let’s walk you to your extraordinary future.\n#ITCANBE\n   

                                                                  Link  \
2  https://linkedin.com/feed/update/urn:li:ugcPost:7084954641788018688   
3    https://linkedin.com/feed/update/urn:li:share:7084529022105231360   
5    https://linkedin.com/feed/update/urn:li:share:7084215735241052161   
6    https://linkedin.com/feed/update/urn:li:share:7084189782313943041   
7    https://linkedin.com/feed/update/urn:li:share:7084165117235388416   

                  Tags  
2  Weekly Wealth Posts  
3  Weekly Wealth Posts  
5  Enterprise Campaign  
6                  BCC  
7  Weekly Wealth Posts

In [22]:
# Check value counts of tags
df_not_null['Tags'].value_counts()

CIB Posts                                234
Weekly Wealth Posts                      189
CHNW Mass Retail                         122
CHNW Mass Retail, Reward4Saving           94
YLS                                       92
                                        ... 
BCC, LetUsRebuildTogether                  1
Trustees Posts                             1
CHNW Mass Retail, Infrastructure Fund      1
BCC Thematic Campaign                      1
wedding series                             1
Name: Tags, Length: 94, dtype: int64

##### Note

Missing tags will be imputed with blank space.

In [23]:
# fill missing values in 'Post' column with values in ''
df_not_null['Tags'] = df_not_null['Tags'].fillna('')

In [24]:
# Check missing values
pd.DataFrame(df_not_null.isna().sum()).sort_values(by=0, ascending=False).head()

0
Date                              0
Engagement Rate (per Impression)  0
Post Clicks (All)                 0
Post Link Clicks                  0
Click-Through Rate                0

In [25]:
# Check Impressions and 'Organic Impressions' for differences
df_not_null[(df_not_null['Impressions'].notna()) &
    (df_not_null['Impressions'] != df_not_null['Organic Impressions'])
            ][['Impressions', 'Organic Impressions']]

Empty DataFrame
Columns: [Impressions, Organic Impressions]
Index: []

----

##### Additional columns to be dropped


* **'Post ID'**: There is no reason to track individual posts in this analysis.
* **Network, 'Post Type', 'Profile**: Redundant Information
* **Link**: The actual link to linkedin posts are not needed for analysis
* **Organic Impressions**: This column has the same data as the Impression column

In [26]:
# Drop additional columns 
cols = ['Post ID', 'Network', 'Post Type', 'Profile','Link', 'Organic Impressions']

df_not_null = df_not_null.drop(cols, axis=1)

In [27]:
# Check missing values
pd.DataFrame(df_not_null.isna().sum()).sort_values(by=0, ascending=False)

0
Date                              0
Content Type                      0
Sent by                           0
Post                              0
Impressions                       0
Engagement Rate (per Impression)  0
Engagements                       0
Reactions                         0
Likes                             0
Comments                          0
Shares                            0
Click-Through Rate                0
Post Link Clicks                  0
Post Clicks (All)                 0
Tags                              0

---
#### Fix issues with 'Engagement Rate (per impression)', 'Click-Through Rate' Formatting

In [28]:
# Remove % sign from 'Engagement Rate (per Impression)', 'Engagement Rate (per Reach)' & 'Click-Through Rate' columns
df_not_null['Engagement Rate (per Impression)'] = df_not_null['Engagement Rate (per Impression)'].str.replace('%', '')
df_not_null['Click-Through Rate'] = df_not_null['Click-Through Rate'].str.replace('%', '')

In [29]:
# Confirm changes
df_not_null[['Engagement Rate (per Impression)', 'Click-Through Rate']].head()

Engagement Rate (per Impression) Click-Through Rate
2                             9.52               4.76
3                             1.83               1.22
4                             5.05               2.13
5                             2.04               2.04
6                             4.78               2.39

---
#### Fix issues with wrong data types

In [30]:
# Convert "Date" column to datetime
df_not_null['Date'] = pd.to_datetime(df_not_null['Date'])

In [31]:
df_not_null.columns

Index(['Date', 'Content Type', 'Sent by', 'Post', 'Impressions',
       'Engagement Rate (per Impression)', 'Engagements', 'Reactions', 'Likes',
       'Comments', 'Shares', 'Click-Through Rate', 'Post Link Clicks',
       'Post Clicks (All)', 'Tags'],
      dtype='object')

In [32]:
df_not_null.dtypes

Date                                datetime64[ns]
Content Type                                object
Sent by                                     object
Post                                        object
Impressions                                 object
Engagement Rate (per Impression)            object
Engagements                                 object
Reactions                                  float64
Likes                                      float64
Comments                                   float64
Shares                                     float64
Click-Through Rate                          object
Post Link Clicks                            object
Post Clicks (All)                           object
Tags                                        object
dtype: object

In [33]:
df_not_null.columns

Index(['Date', 'Content Type', 'Sent by', 'Post', 'Impressions',
       'Engagement Rate (per Impression)', 'Engagements', 'Reactions', 'Likes',
       'Comments', 'Shares', 'Click-Through Rate', 'Post Link Clicks',
       'Post Clicks (All)', 'Tags'],
      dtype='object')

In [34]:
# List of columns to convert to numeric
numeric_columns =  ['Impressions', 'Engagement Rate (per Impression)', 'Engagements', 'Click-Through Rate', 'Post Link Clicks',
       'Post Clicks (All)']

In [35]:
# Convert columns to numeric data type
df_not_null[numeric_columns] = df_not_null[numeric_columns].applymap(lambda x: float(str(x).replace(',', '')))

In [36]:
# Convert 'Sent by' and 'Content type' columns to categorical data type
df_not_null['Sent by'] = df_not_null['Sent by'].astype('category')
df_not_null['Content Type'] = df_not_null['Content Type'].astype('category')

In [37]:
# Examine datatypes
pd.DataFrame(df_not_null.dtypes)

0
Date                              datetime64[ns]
Content Type                            category
Sent by                                 category
Post                                      object
Impressions                              float64
Engagement Rate (per Impression)         float64
Engagements                              float64
Reactions                                float64
Likes                                    float64
Comments                                 float64
Shares                                   float64
Click-Through Rate                       float64
Post Link Clicks                         float64
Post Clicks (All)                        float64
Tags                                      object

---

#### Fix issues with 'Sent by' blank spaces

In [38]:
pd.DataFrame(df_not_null['Sent by'].value_counts())

Sent by
Blessing Ubah         2348
                      1709
Sophia Amuka          1573
Kemi Amoo              327
Aramide Salami         268
Damilare Oyekanmi       96
Dunni Ogunsanya         11

In [39]:
# Replace blank space with 'Other' in 'Sent by' column
df_not_null['Sent by'] = df_not_null['Sent by'].replace(' ', 'Other')

In [40]:
# Confirm changes
pd.DataFrame(df_not_null['Sent by'].value_counts())

Sent by
Blessing Ubah         2348
Other                 1709
Sophia Amuka          1573
Kemi Amoo              327
Aramide Salami         268
Damilare Oyekanmi       96
Dunni Ogunsanya         11

---